ext/xsl/xsltprocessor.c

```diff
@@ -574,26 +574,34 @@ PHP_METHOD(XSLTProcessor, transformToDoc)
 			}
 
 			ce = zend_lookup_class(ret_class);
-			if (ce == NULL || !instanceof_function(ce, curce)) {
-				xmlFreeDoc(newdocp);
+			if (ce == NULL) {
+				zend_argument_type_error(2, "must be a valid class");
+				goto fail;
+			}
+			if (!instanceof_function(ce, curce)) {
```

Comment:
Can't the ``C`` ZPP type specifier be used instead for this?

Reply:
TIL, thanks. I've updated this to use C.